frontend/service.go

```diff
@@ -28,94 +21,110 @@ var (
 )
 
 // the service type
-type SqlStreamer struct {
-	db     *sql.DB
+type LwdStreamer struct {
+	cache  *common.BlockCache
 	client *rpcclient.Client
 	log    *logrus.Entry
 }
 
-func NewSQLiteStreamer(dbPath string, client *rpcclient.Client, log *logrus.Entry) (walletrpc.CompactTxStreamerServer, error) {
-	db, err := sql.Open("sqlite3", fmt.Sprintf("file:%s?_busy_timeout=10000&cache=shared", dbPath))
-	db.SetMaxOpenConns(1)
-	if err != nil {
-		return nil, err
-	}
+func NewLwdStreamer(client *rpcclient.Client, cache *common.BlockCache, log *logrus.Entry) (walletrpc.CompactTxStreamerServer, error) {
+	return &LwdStreamer{cache, client, log}, nil
+}
 
-	// Creates our tables if they don't already exist.
-	err = storage.CreateTables(db)
-	if err != nil {
-		return nil, err
+func (s *LwdStreamer) GetCache() *common.BlockCache {
+	return s.cache
+}
+
+func (s *LwdStreamer) GetLatestBlock(ctx context.Context, placeholder *walletrpc.ChainSpec) (*walletrpc.BlockID, error) {
+	latestBlock := s.cache.GetLatestBlock()
+
+	if latestBlock == -1 {
+		return nil, errors.New("Cache is empty. Server is probably not yet ready.")
 	}
 
-	return &SqlStreamer{db, client, log}, nil
+	// TODO: also return block hashes here
+	return &walletrpc.BlockID{Height: uint64(latestBlock)}, nil
 }
 
-func (s *SqlStreamer) GracefulStop() error {
-	return s.db.Close()
-}
+func (s *LwdStreamer) GetAddressTxids(addressBlockFilter *walletrpc.TransparentAddressBlockFilter, resp walletrpc.CompactTxStreamer_GetAddressTxidsServer) error {
+	params := make([]json.RawMessage, 1)
+	st := "{\"addresses\": [\"" + addressBlockFilter.Address + "\"]," +
```

Comment:
@defuse Identified a security problem here where the address field could contain multiple addresses, and cause a DoS attack. 

See his fix here: https://github.com/adityapk00/lightwalletd/pull/4/files

Reply:
Please see the comment I just wrote on that PR: https://github.com/adityapk00/lightwalletd/pull/4#issuecomment-566841523

Reply:
Note that I just cherry-picked this commit: https://github.com/adityapk00/lightwalletd/commit/cc30455a72a8d1fa79bcb160eaf7ef87e3a0720d (but I didn't include the later correction to replace the `^` and `$` from the pattern because I don't think that's necessary).

Reply:
I left a comment on that PR (https://github.com/adityapk00/lightwalletd/pull/4#issuecomment-567122220) saying `\A` and `\z` make it more obvious nothing's wrong, but this PR doesn't have to block on that.